frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineRunTaskRunGraph.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { DEFAULT_CHART_HEIGHT, DEFAULT_LEGEND_CHART_HEIGHT } from '../const';
+import {
+  ChartLegend,
+  ChartVoronoiContainer,
+  getInteractiveLegendEvents,
+  getInteractiveLegendItemStyles,
+} from '@patternfly/react-charts';
+import { GraphEmpty } from '@console/internal/components/graphs/graph-empty';
+import { formatDuration } from '@console/internal/components/utils/datetime';
+import { LoadingInline, truncateMiddle } from '@console/internal/components/utils';
+import { usePipelineRunTaskRunPoll } from '../hooks';
+import {
+  PipelineMetricsGraphProps,
+  getRangeVectorData,
+  getYaxisValues,
+} from './pipeline-metrics-utils';
+import { LineChart } from './charts/lineChart';
+
+import './pipeline-chart.scss';
+
+const PipelineRunTaskRunGraph: React.FC<PipelineMetricsGraphProps> = ({
+  pipeline,
+  timespan,
+  interval,
+  width = 650,
+}) => {
+  const {
+    metadata: { name, namespace },
+  } = pipeline;
+  const { t } = useTranslation();
+  const [hiddenSeries, setHiddenSeries] = React.useState<Set<number>>(new Set());
+  const chartHeight = DEFAULT_LEGEND_CHART_HEIGHT;
+  const [runData, runDataError, runDataLoading] = usePipelineRunTaskRunPoll({
+    name,
+    namespace,
+    timespan,
+    delay: interval,
+  });
+
+  if (runDataLoading) {
+    return <LoadingInline />;
+  }
+
+  const pipelineRunResultData = runData?.data?.result ?? [];
+  if (runDataError || pipelineRunResultData.length === 0) {
+    return <GraphEmpty height={chartHeight - 30} />;
+  }
+
+  const pRuns = getRangeVectorData(runData, (r) => truncateMiddle(r.metric.pipelinerun)) ?? [];
```

Comment:
I think we want to truncate more to get us down to a small bitesized item... we definitely need that tooltip improvement though so they can see the full PipelineRun name.
```suggestion
  const pRuns = getRangeVectorData(runData, (r) => truncateMiddle(r.metric.pipelinerun, { length: 10 })) ?? [];
```